runtime/native/unsafe.cpp

```diff
@@ -277,9 +287,25 @@ Java_sun_misc_Unsafe_staticFieldOffset(JNIEnv *env, jobject, jobject) {
     UnsafeUnimplemented("Java_sun_misc_Unsafe_staticFieldOffset");
 }
 
+unsigned int fieldSlotOffset = -1;
 jlong
-Java_sun_misc_Unsafe_objectFieldOffset(JNIEnv *env, jobject, jobject) {
-    UnsafeUnimplemented("Java_sun_misc_Unsafe_objectFieldOffset");
+Java_sun_misc_Unsafe_objectFieldOffset(JNIEnv *env, jobject unsafeObj, jobject fieldObj) {
+    // dw475 TODO check back
+    long offset = 0;
+    if (fieldSlotOffset == -1) {
+        const JavaClassInfo* info = GetJavaClassInfo(Unwrap(fieldObj)->Cdv()->Class()->Wrap());
+        for (int i = 0; i < info->num_fields; i++) {
+            if (strcmp(info->fields[i].name, "slot") == 0) {
+                offset = *((jint *)(((char *) fieldObj)+info->fields[i].offset));
+                break;
+            }
+        }
+    } else {
+        offset = *((jint *)(((char *) fieldObj)+fieldSlotOffset));
+    }
+    return offset;
+
+    // return *((jlong*)(((char*) fieldObj)+48));
```

Comment:
You should also implement the `staticFieldBase(...)` and `staticFieldOffset(...)`functions. The former can totally just be `return null` for now, since we have global pointers to static fields.